app/controllers/libraries_controller.rb

```diff
@@ -97,10 +97,10 @@ def destroy
 
   # Never trust parameters from the scary internet, only allow the white list through.
   def library_params
-    params.require(:library).permit(:name, :slug, node_mapping_attributes: [:node_id], locations_attributes: locations_params)
+    params.require(:library).permit(:name, :slug, node_mapping_attributes: [:id, :node_id], locations_attributes: locations_params)
```

Comment:
I'm not sure we have to be super worried about this, but will this theoretically allow somebody to edit any library?

Reply:
(Other than we currently let every admin edit everything..), I'm pretty sure Rails also has an internal check that the relationship is on that library.

Reply:
And double-checked:

```
irb(main):007:0> Library.first.update(node_mapping_attributes: { id: 3, node_id:321 })
ActiveRecord::RecordNotFound (Couldn't find NodeMapping with ID=3 for Library with ID=1)
```